.github/workflows/build-and-test.yml

```diff
@@ -4,15 +4,20 @@ on: [push, pull_request]
 
 jobs:
   build-and-test:
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        java: [8, 8.0.192, 11, 11.0.9, 12, 13, 13.0.4, 14.0.2, 15]
+        os: [ubuntu-latest, macOS-latest]
```

Comment:
why not windows?

Reply:
I ran out of time, and left it out because the steps to remove the artifacts from the cache (.m2 area) is using unix/linux commands such as `rm -rf` as opposed to `rmdir` or `del` commands in Windows.

Reply:
Mark, 
I removed the JDKs that are EOL.  Let me know what you think. 
-Carl

Reply:
ultimately it's not my call but LGTM, 
I'll leave it to @jnh5y as the last committer on the workflows